backends/presence/discord/discord.cpp

```diff
@@ -29,19 +36,85 @@
 #include <discord_rpc.h>
 #include <time.h>
 
+#ifdef WIN32
+#include <windows.h>
+#else
+#include <dlfcn.h>
+#endif
+
 #define DISCORD_CLIENT_ID "714287866464698470"
 
+static void (DISCORD_CALL *Discord_Initialize_dyn)(const char* applicationId,
+                                       DiscordEventHandlers* handlers,
+                                       int autoRegister,
+                                       const char* optionalSteamId);
+static void (DISCORD_CALL *Discord_Shutdown_dyn)(void);
+static void (DISCORD_CALL *Discord_UpdatePresence_dyn)(const DiscordRichPresence* presence);
+static void (DISCORD_CALL *Discord_ClearPresence_dyn)(void);
+
+void loadDiscordLib() {
+	static bool attempted = false;
+	if (attempted)
+		return;
+	attempted = true;
+#ifdef WIN32
+	HINSTANCE handle = LoadLibrary(TEXT("discord-rpc.dll"));
+	if (!handle) {
+		warning("Failed to load discord-rpc.dll");
+		return;
+	}
+	*(void **) (&Discord_Initialize_dyn) = GetProcAddress(handle, "Discord_Initialize");
+	*(void **) (&Discord_Shutdown_dyn) = GetProcAddress(handle, "Discord_Shutdown");
+	*(void **) (&Discord_UpdatePresence_dyn) = GetProcAddress(handle, "Discord_UpdatePresence");
+	*(void **) (&Discord_ClearPresence_dyn) = GetProcAddress(handle, "Discord_ClearPresence");
+#else
+	void *handle = dlopen("discord-rpc.so", RTLD_NOW | RTLD_GLOBAL);
```

Comment:
On macOS the library is called `libdiscord-rpc.dylib`, not `discord-rpc.so`.
Also when building a bundle, the dynamic libraries are not in the same place as the executable. How is it going to find it? Usually this is done by setting the rpath (see how Sparkle is handed in `ports.mk`), but that only works for libraries linked to the executable.

Maybe it would be better to limit the change to Windows if the issue you are trying to fix only exists there?